packages/itwin/reports-config-widget/src/widget/components/ExtractionStatus.tsx

```diff
@@ -0,0 +1,150 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { ProgressRadial } from "@itwin/itwinui-react";
+import * as React from "react";
+import { useEffect, useState } from "react";
+import "./ExtractionStatus.scss";
+import {
+  SvgStatusError,
+  SvgStatusPending,
+  SvgStatusPendingHollow,
+  SvgStatusSuccess,
+} from "@itwin/itwinui-icons-color-react";
+import { ReportsConfigWidget } from "../../ReportsConfigWidget";
+
+export enum ExtractionStates {
```

Comment:
You might want to extend extraction status enum that Extraction API returns. Take a look at this: https://blog.logrocket.com/extend-enums-typescript/

This might make the conversions easier.

Reply:
`ExtractionStates` enum has not been extended as conversion from `ExtractorState` is being done in `getFinalState` in `BulkExtractor`

Reply:
Typescript does not support simple enum extensions. Extending them will cause more problems.